### PR TITLE
Prohibit use of '/#/' in callback URL

### LINKED ISSUE #401: Cannot have /#/ in callback url
`http://localhost:8080/#/callback`  can't be used as valid redirect uri. However the following will work,

- `http://localhost:8080/callback`
- `http://localhost:8080/xyz/callback`

### PR DESCRIPTION
**Pull request by AdityaHirapara**
Fixes #401.
oneauth_callback

### CODE REVIEW

**Comment by championswimmer** [comment]
check in backend and req.flash()

except the profile photo upload, we try to use minimal frontend javascript

**Comment by abhishek97** [comment]
Can't we support it? Why prohibit it?

**Comment by AdityaHirapara** [comment]
@abhishek97, what do you mean by supporting?

**Comment by abhishek97** [comment]
Can we allow it? 
Why is it not working when we have /#/ ? Can we make changes such that the auth will work even for /#/ callbacks?

**Comment by AdityaHirapara** [comment]
We can support it, but it will create lots of confusion.

Problem here is, when we generate client with callback url like `"http://example.com/#/callback"` and when any user log in through that client, it will redirect him to url `"http://localhost:3838/oauth/authorize?response_type=code&client_id=7638580675&redirect_uri=http://example.com/#/callback"`

Here in oneauth, library we are using for OAuth authorization will neglect URL after #, so it will take callback URL as `"http://example.com/"` and then we are trying to check it with `"http://example.com/#/callback"` from database, which is creating a problem and giving unauthorized client error.

We can solve this by removing URL segment after # in database, before checking it. Then it won't give unauthorized client error.

But it does not have any worth here, as URL segment after # always will be neglected, even client can't define any route like `"/#/callback"` in his app, it will simply be executed as `"/"`. So prohibiting it will be a good idea, as it will remove lots of confusion.

**Comment by championswimmer** [comment]
yes, the reasoning is correct @AdityaHirapara ! 

i am on board with the reason, but see if you can do via backend validation and req.flash() instead

**Comment by AdityaHirapara** [comment]
Yes, I have almost done with that.

File: src/routers/api/clients.js [hunk 1]
```diff
@@ -30,6 +30,12 @@ router.post('/add', async function (req, res) {
         return res.status(403).send("Only logged in users can make clients")
     }
 
+    let hashRegEx = /\/#\//
+    if (hashRegEx.test(req.body.callback)) {
+        req.flash('error', '"/#/" is not allowed in Callback URL')
+        return res.redirect('/users/me')
```
[the remark this answers is not in the document]
Comment (AdityaHirapara) [reply src/routers/api/clients.js]: Done!

**Comment by championswimmer** [comment]
Awesome !

**Pull request merged**